### PR TITLE
Atomic Revival: Display My Home notices for all three stages of revival

### PR DESCRIPTION
**Pull request by p-jackson**
See https://github.com/Automattic/dotcom-forge/issues/1096

## Proposed Changes

Displays a My Home notice for each stage of Atomic Revival. They will go live once D90984-code is committed.

### Stage 1: Site automatically reverted, has no plan

CleanShot 2022-11-01 at 07 27 51@2x

### Stage 2: Site automatically reverted, Business plan or higher added (lower plan wouldn't display)

image

### Stage 3: Revived to Atomic

CleanShot 2022-11-01 at 07 30 31@2x

## Testing Instructions

1. Apply D90984-code and sandbox `public-api.wordpress.com`.
2. Create a new Free site.
3. Mock Stage 1 by adding `woa-revert-from-auto-cleanup` blog sticker: `wp blog-stickers add --sticker=woa-revert-from-auto-cleanup --who=<who> --blog_id=<id>`.
4. Visit My Home and verify the Stage 1 notice appears.
5. Purchase a Business plan for the site.
6. Visit My Home and verify the Stage 2 notice appears.
7. Activate Hosting features.
8. Visit My Home and verify the Stage 3 notice appears.
9. Restore a backup.
10. Visit My Home and verify the notice no longer appears.

### CODE REVIEW

**Review by mpkelly — APPROVED** [review]
Looks good to me. I tested with all Calypso themes on Chrome and Modern theme on Firefox.

**Comment by p-jackson** [comment]
Thanks @mpkelly. I'm going to wait and see if there's any feedback on the wording.

File: client/my-sites/customer-home/cards/tasks/reactivate-expired-plan/index.tsx [hunk 1]
```diff
@@ -26,6 +28,20 @@ export const ReactivateExpiredPlan = () => {
...
+								supportLink={ localizeUrl(
+									'https://wordpress.com/support/restore-your-site-after-the-plan-expires/'
+								) }
+								noWrap={ false }
+								showIcon={ false }
+								supportContext="reactivate"
+								tracksEvent="calypso_customer_home_reactivate_support_page_view"
+								statsGroup="calypso_customer_home"
+								statsName="menus_view_tutorial"
+							/>
```
Comment (sejas) [inline client/my-sites/customer-home/cards/tasks/reactivate-expired-plan/index.tsx]: I didn't know this component `InlineSupportLink`.  It looks very handy.

File: client/my-sites/customer-home/cards/tasks/style.scss [hunk 2]
```diff
@@ -128,7 +128,7 @@
 
 		.task__description a {
 			color: var(--color-neutral-0);
-			font-weight: bold;
+			text-decoration: underline;
```
Comment (sejas) [inline client/my-sites/customer-home/cards/tasks/style.scss:131]: Maybe we could add a hover effect.

**Review by sejas — APPROVED** [review]
Looking good! I've learnt about `InlineSupportLink` and how the expire auto-revert process work by reading the support page 👌 .
I've added a suggestion about adding a hover effect into the link style.

**Review by danielbachhuber — COMMENTED** [review]
This is really cool! I love the inline support doc.

What if we _reused_ this notice for all three stages of the Atomic Revival? My assumptions are:
1. If we reuse the notice, they can dismiss it once and never see it again.
2. We can have the notice adapt based on which stage they're in.

The button for each step would take them to the page where they need to perform the action.

We can also do the three stage logic in a follow-up PR.

## Stage 1: Reverted, no plan

image

## Stage 2: Reverted, with Business plan or higher (lower plan wouldn't display)

image

## Stage 3: Revived to Atomic

image

**Comment by p-jackson** [comment]
Thanks @danielbachhuber, I'm going to try this out. The logic will be here D90984-code, but I'm thinking of having a UI something like this

**Step 1**
CleanShot 2022-10-28 at 17 56 02@2x

**Step 2**
CleanShot 2022-10-28 at 17 56 43@2x

**Step 3**
CleanShot 2022-10-28 at 17 57 19@2x

**Notes**
- I've toned down the banners to the non-urgent style. We could also have step 1 be urgent and the rest non-urgent
- I _didn't_ include "Step N" text in the button label. That's because users may not necessarily have gone through step 1 before seeing the step 2 banner, which could be confusing i.e. it's possible the user upgraded to a business plan through one of the other Atomic Revival mechanisms we've added during this project, so when they land on My Home they'll jump straight to step 2.
- Step 2 and 3 are much easier to permanently dismiss. It's really not a problem if they don't want to transfer to Atomic, and it's simpler for us too.
- The support document has a step 4 for launching the site. I might include this step too, but there could be some trickiness about whether the site is actually un-launched or just in coming soon mode.

**Comment by mpkelly** [comment]
I retested this after applying D90984-code and think it looks good. As we are back to using the Calypso theme colours for links and buttons, I tried with all of them and found it was ok now we are using the white background. The worst one is the Contrast theme, ironically, but it is still passable, I think.  

So we ditched the inline help and went with external links?

**Comment by p-jackson** [comment]
> So we ditched the inline help and went with external links?

@mpkelly yeah this is something @danielbachhuber and I chatted about on a huddle. We want to lean on the step-by-step process in the support docs, so we feel it's better to encourage the user to leave it open in another tab.

**Comment by danielbachhuber** [comment]
> I'm thinking of having a UI something like this

@p-jackson This is really excellent work. Nice job 💪

**Comment by danielbachhuber** [comment]
Some minor Reactor feedback pdKhl6-GA-p2#comment-1045

**Comment by p-jackson** [comment]
CleanShot 2022-10-31 at 17 46 01@2x

I do like the idea of having something like this in the card. It sets expectations with the user that this is going to be a process—not a single click and you're done.

Spent most of today on D90984-code, figuring out the best way to decide what step the user is up to.

**Review by danielbachhuber — APPROVED** [review]
Great work on this, @p-jackson. Appreciate you diving into it!

You're going to hate me but I restored the `isUrgent` prop. Because we _automatically_ reverted their site, I think we need to make it as obvious as we can. They can easily dismiss it.

I updated the description with new screenshots. I also registered the Tracks event https://github.com/Automattic/tracks-events-registration/pull/1158

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7606919

Thank you @p-jackson for including a screenshot in the description! This is really helpful for our translators.

**Comment by p-jackson** [comment]
I've updated the illustration to use the dark variant https://github.com/Automattic/wp-calypso/pull/69678

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7606919).

**Pull request merged**